### PR TITLE
DHFPROD-9750: Use real data for Create, Edit and copy lists

### PR DESCRIPTION
**Pull request by amejiaq**
#### Checklist: 
```diff
- Note: do not change the below
```

- ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
- [x] Ran cypress tests on firefox locally
  

- ##### Reviewer:

- [x] Reviewed Tests
- [x] Added to Release Wiki

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/api/__mocks__/mocks.data.ts [hunk 1]
```diff
@@ -361,6 +361,20 @@ const clearUserDataAPI = (axiosMock) => {
   });
 };
 
+const getAllExcludeValuesList = (axiosMock) => {
+  return  axiosMock.get["mockImplementation"]((url) => {
+    switch (url) {
+    case "/api/steps/matching/exclusionList":
+      return Promise.resolve({
+        "data": {"name": "Preset List 1", "values": ["uno", "dos", "uno", "dos", "uno", "dos", "uno", "dos"]},
```
Comment (littlecherro) [inline marklogic-data-hub-central/ui/src/api/__mocks__/mocks.data.ts:369]: i get what you meant to do here but i think we should stick to the same language through the app :)

File: marklogic-data-hub-central/ui/src/components/entities/matching/list-modal/list-modal.test.tsx [hunk 2]
```diff
@@ -31,14 +39,38 @@ describe("Test input validation values to ignore", () => {
...
     });
-    render(<ListModal isVisible={true} toggleModal={() => {}} action={"C"} confirmAction={() => {}} />);
+    render(<ListModal
+      isVisible={true}
+      toggleModal={() => {}}
+      action={"A"}
+      confirmAction={() => {}}
+      checkIfExistInList={() => { return false; }}
+      updateListValues={() => {}}
+      listValues={["Uno"]}
```
Comment (littlecherro) [inline marklogic-data-hub-central/ui/src/components/entities/matching/list-modal/list-modal.test.tsx:49]: same here :)

File: marklogic-data-hub-central/ui/src/components/entities/matching/list-modal/list-modal.test.tsx [hunk 3]
```diff
@@ -31,14 +39,38 @@ describe("Test input validation values to ignore", () => {
...
+
+  it("Don't allow List Name duplicated", () => {
+    render(<ListModal
+      isVisible={true}
+      toggleModal={() => {}}
+      action={"A"}
+      confirmAction={() => {}}
+      checkIfExistInList={() => { return true; }}
+      updateListValues={() => {}}
+      listValues={["Uno"]}
```
Comment (littlecherro) [inline marklogic-data-hub-central/ui/src/components/entities/matching/list-modal/list-modal.test.tsx:68]: here too

File: marklogic-data-hub-central/ui/src/components/entities/matching/ruleset-single-modal/ruleset-single-modal.test.tsx [hunk 4]
```diff
@@ -425,7 +436,21 @@ describe("Matching Ruleset Single Modal component", () => {
...
 
-  test("Hover on Values to Ignore items", async () => {
+  test("Render list to ignore and tooltip when hover list items", async () => {
+    jest.resetAllMocks();
+    mockGetAllExcludeValuesList.mockResolvedValue(
+      {
+        data: [
+          {
+            "name": "Preset List 1",
+            "values": ["uno", "dos", "uno", "dos", "uno", "dos", "uno", "dos"]
```
Comment (littlecherro) [inline marklogic-data-hub-central/ui/src/components/entities/matching/ruleset-single-modal/ruleset-single-modal.test.tsx:446]: :)

File: marklogic-data-hub-central/ui/src/components/entities/matching/list-modal/list-modal.test.tsx (hunk 2 above)
Comment (amejiaq) [reply marklogic-data-hub-central/ui/src/components/entities/matching/list-modal/list-modal.test.tsx]: Done!

File: marklogic-data-hub-central/ui/src/components/entities/matching/ruleset-single-modal/ruleset-single-modal.test.tsx (hunk 4 above)
Comment (amejiaq) [reply marklogic-data-hub-central/ui/src/components/entities/matching/ruleset-single-modal/ruleset-single-modal.test.tsx]: Done!

**Review by MSBrignoni — CHANGES_REQUESTED** [review]
Looks good! there is only this little detail, perhaps is possible to control this in the front:



https://user-images.githubusercontent.com/89084298/213202956-c1a56d69-b359-432f-b1d7-e9dfe7b3a5cd.mp4

**Comment by amejiaq** [comment]
> Looks good! there is only this little detail, perhaps is possible to control this in the front:
> 
>  Screen_Recording_.1-18-2023_11-44-29_AM.mp4

This behavior will be review it in this ticket https://project.marklogic.com/jira/browse/DHFPROD-9811

**Review by MSBrignoni — APPROVED** [review]

**Review by littlecherro — APPROVED** [review]

**Pull request merged**